### ISSUE TITLE
Slider relies on brittle paint order to work properly

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#98770]
The material Slider widget relies on the order in which nodes get precessed during PAINT in order to present itself visually correctly. That's brittle.

Here's the problem: In its paint method `_RenderSlider` configures the `paintValueIndicator` painter... 

https://github.com/flutter/flutter/blob/5684b9f69e86712107b14c39dcbcd16c7d4dc08f/packages/flutter/lib/src/material/slider.dart#L1453

... which is used in the paint method of `_RenderValueIndicator`:

https://github.com/flutter/flutter/blob/5684b9f69e86712107b14c39dcbcd16c7d4dc08f/packages/flutter/lib/src/material/slider.dart#L1611

This only works if `_RenderSlider` is processed before `_RenderValueIndicator` during PAINT. Otherwise, `_RenderValueIndicator.paint` will not see the painter configured by `_RenderSlider.paint` and instead paint with an outdated painter.

Since nodes are processed from deepest to shallowest, this works fine when `_RenderSlider` is deeper in the tree than `_RenderValueIndicator` and breaks otherwise. It will also break when the order in which we process nodes during PAINT is flipped around as we did in https://github.com/flutter/flutter/pull/98219, which may give us some performance benefits. That PR saw breakages in google3 because of the problem described here (see b/219153804). 

Long story short: Slider should not relay on the somewhat brittle order in which nodes get processed during PAINT.

~Side node: We should also add a test for this to the framework since this was only caught during a google3 roll.~ Done: https://github.com/flutter/flutter/pull/98772

**Comment by goderbauer** [comment]
That behavior was introduced in https://github.com/flutter/flutter/pull/53916. @HansMuller @JoseAlba @clocksmith

**Comment by TahaTesser** [comment]
@goderbauer 
I wanna pick your brain regarding `Slider. I noticed `Slider` uses `FocusableActionDetector` on entire `_SliderRenderObjectWidget`, which paints tracks, thumb and value indicator together.

 I wanna be able to detect hover/focus just on my thumb to update overlay and as well as to add support thumb overlays in `RangeSlider`.


I was thinking instead, I could use `CustomMultiChildLayout` to render all these components individually with `LayoutId`. 

Something like this:
```dart
        child: CompositedTransformTarget(
          link: _layerLink,
          child: CustomMultiChildLayout(
            delegate: _SliderLayoutDelegate(),
            children: <Widget>[
              LayoutId(
                id: _SliderChildSlot.sliderTrack,
                child: _SliderTrackObjectWidget(
                  ...
                ),
              ),
              LayoutId(
                id: _SliderChildSlot.sliderThumb,
                child: _SliderVIndicatorObjectWidget(
                  ...
                ),
              ),
              LayoutId(
                id: _SliderChildSlot.sliderValueIndicator,
                child: _SliderValueIndicatorObjectWidget(
                  ...
                ),
              ),
            ],
          ),
        ),
```


> This only works if _RenderSlider is processed before _RenderValueIndicator during PAINT. Otherwise, _RenderValueIndicator.paint will not see the painter configured by _RenderSlider.paint and instead paint with an outdated painter.

Would this also help this issue? Wouldn't this also ensure `_RenderSlider` is rendered before `_RenderValueIndicator`?

**Comment by TahaTesser** [comment]
cc: @HansMuller

**Comment by TahaTesser** [comment]
cc: @darrenaustin
